chronos/apps/http/httpcommon.nim

```diff
@@ -0,0 +1,199 @@
+    ok(res)
+  else:
+    for header in ch:
+      for item in header.split(","):
+        case strip(item.toLowerAscii())
+        of "identity":
+          res.incl(TransferEncodingFlags.Identity)
+        of "chunked":
+          res.incl(TransferEncodingFlags.Chunked)
+        of "compress":
+          res.incl(TransferEncodingFlags.Compress)
+        of "deflate":
+          res.incl(TransferEncodingFlags.Deflate)
+        of "gzip":
+          res.incl(TransferEncodingFlags.Gzip)
+        of "x-gzip":
+          res.incl(TransferEncodingFlags.Gzip)
+        of "":
+          res.incl(TransferEncodingFlags.Identity)
+        else:
+          return err("Incorrect Transfer-Encoding value")
+    ok(res)
+
+func getContentEncoding*(ch: openarray[string]): HttpResult[
+                                                   set[ContentEncodingFlags]] {.
+     raises: [Defect].} =
+  ## Parse value of multiple HTTP headers ``Content-Encoding`` and return
+  ## it as set of ``ContentEncodingFlags``.
+  var res: set[ContentEncodingFlags] = {}
+  if len(ch) == 0:
+    res.incl(ContentEncodingFlags.Identity)
+    ok(res)
+  else:
+    for header in ch:
+      for item in header.split(","):
+        case strip(item.toLowerAscii()):
+        of "identity":
+          res.incl(ContentEncodingFlags.Identity)
+        of "br":
+          res.incl(ContentEncodingFlags.Br)
+        of "compress":
+          res.incl(ContentEncodingFlags.Compress)
+        of "deflate":
+          res.incl(ContentEncodingFlags.Deflate)
+        of "gzip":
+          res.incl(ContentEncodingFlags.Gzip)
+        of "x-gzip":
+          res.incl(ContentEncodingFlags.Gzip)
+        of "":
+          res.incl(ContentEncodingFlags.Identity)
+        else:
+          return err("Incorrect Content-Encoding value")
+    ok(res)
+
+func getContentType*(ch: openarray[string]): HttpResult[string]  {.
+     raises: [Defect].} =
+  ## Check and prepare value of ``Content-Type`` header.
+  if len(ch) > 1:
+    err("Multiple Content-Type values found")
+  else:
+    let mparts = ch[0].split(";")
+    ok(strip(mparts[0]).toLowerAscii())
+
+proc bytesToString*(src: openarray[byte], dst: var openarray[char]) =
+  ## Convert array of bytes to array of characters.
+  ##
+  ## Note, that this procedure assume that `sizeof(byte) == sizeof(char) == 1`.
+  ## If this equation is not correct this procedures MUST not be used.
+  doAssert(len(src) == len(dst))
+  if len(src) > 0:
+    copyMem(addr dst[0], unsafeAddr src[0], len(src))
+
+proc stringToBytes*(src: openarray[char], dst: var openarray[byte]) =
+  ## Convert array of characters to array of bytes.
+  ##
+  ## Note, that this procedure assume that `sizeof(byte) == sizeof(char) == 1`.
+  ## If this equation is not correct this procedures MUST not be used.
+  doAssert(len(src) == len(dst))
+  if len(src) > 0:
+    copyMem(addr dst[0], unsafeAddr src[0], len(src))
+
+func bytesToString*(src: openarray[byte]): string =
```

Comment:
anything wrong with `stew/byteutils`? there's also `toOpenArrayByte()` in system.nim